frontend/src/store/index.js

```diff
@@ -456,12 +456,26 @@ const getters = {
   },
   machineTypesByCloudProfileName (state, getters) {
     return ({ cloudProfileName }) => {
-      return getters.machineTypesByCloudProfileNameAndRegionAndZones({ cloudProfileName })
+      return getters.machineTypesOrVolumeTypesByCloudProfileNameAndRegionAndZones({ type: 'machineTypes', cloudProfileName })
     }
   },
-  machineTypesByCloudProfileNameAndRegionAndZones (state, getters) {
+  machineTypesByCloudProfileNameAndRegionAndZonesAndArchitecture (state, getters) {
+    return ({ cloudProfileName, region, zones, architecture }) => {
+      let machineTypes = getters.machineTypesOrVolumeTypesByCloudProfileNameAndRegionAndZones({ type: 'machineTypes', cloudProfileName, region, zones })
+      machineTypes = map(machineTypes, machineType => {
+        if (!machineType.architecture) {
+          machineType.architecture = 'amd64' // default if not maintained
+        }
+        return machineType
+      })
+
+      return filter(machineTypes, { architecture })
```

Comment:
```suggestion
      const machineTypes = getters.machineTypesOrVolumeTypesByCloudProfileNameAndRegionAndZones({ 
        type: 'machineTypes', 
        cloudProfileName, 
        region, 
        zones 
      })
      for (const machineType of machineTypes) {
          machineType.architecture ??= 'amd64' // default if not maintained
      }

      return filter(machineTypes, 'architecture')
```

Reply:
Your solution is better readable, especially the logical nullish assignment, I need to change it slightly as we need to operate on a copy as we must not modify store objects in the getter.

Reply:
```
        if (!machineType.architecture) {
          machineType.architecture = 'amd64' // default if not maintained
        }
```
is there a linter for such pattern? :)